[jocelyn-soh] iP

jocelyn-soh opened the pull request:
# Jojo

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Jojo frees your mind of having to remember things you need to do. It's,

 - text-based
 - easy to learn
 - ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jocelyn-soh committed:
Completed Level 0

jocelyn-soh committed:
Edited name for Level-0

jocelyn-soh committed:
Added scanner for Level-1

jocelyn-soh committed:
Removed unnecessary comments for Level-0

jocelyn-soh committed:
Completed Level-1

jocelyn-soh committed:
Updated Level-1

jocelyn-soh committed:
Completed Level-2

jocelyn-soh committed:
Added comments for Level-2

jocelyn-soh committed:
Created Task.java

jocelyn-soh committed:
Added updateStatus method in Task.java

jocelyn-soh committed:
Created setDone, setUndone and toString methods

jocelyn-soh committed:
Fixed bugs in toString method

jocelyn-soh committed:
Created createList method

jocelyn-soh committed:
Completed setDone method

jocelyn-soh committed:
Completed setUndone method

jocelyn-soh committed:
Added Deadline, Event and ToDo classes

jocelyn-soh committed:
Updated Event.java

jocelyn-soh committed:
Completed Level 4

jocelyn-soh committed:
Created test cases and added lines between each action

jocelyn-soh committed:
Added exceptions

jocelyn-soh committed:
Update Duke.java

jocelyn-soh committed:
Squashed bugs

jocelyn-soh committed:
Merge branch 'master' of https://github.com/jocelyn-soh/ip

# Conflicts:
#	src/main/java/Duke.java

jocelyn-soh committed:
Updated Duke.java

jocelyn-soh committed:
Updated Duke.java

jocelyn-soh committed:
Updated test cases

jocelyn-soh committed:
Changed from Task[] to ArrayList<Task>

jocelyn-soh committed:
Updated Duke.java to support deleting tasks, added new test cases

jocelyn-soh committed:
Completed Level-6 and updated test cases

blaukc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+                }
+            }
+
+            System.out.println("-----------------------------------------------------------------------");
```
Maybe you should set this as a constant?

blaukc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+            if (userTxt.equals("list")) {
+                createList(listArr);
+            } else if (userTxt.startsWith("mark")) {
+                try {
```
Perhaps the code for `mark`, `unmark` and `delete` should be put into its own function like `createList`, `createTodo`, etc

blaukc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+            if (test.strip().equals("")) {
+                throw new DukeTaskNoDescException();
+            } else {
+                int splitIdx = userTxt.lastIndexOf("/");
```
Should this be `"/by"` instead of `"/"`? (similar to how `createEvent` does it) 
You could also consider using the `stripTrailing()` method on strings to remove unecessary whitespace!

blaukc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+                int splitIdx = userTxt.lastIndexOf("/");
+                String desc = userTxt.substring(9, splitIdx - 1);
+                String by = userTxt.substring(splitIdx + 3);
+                Task t = new Deadline(desc, by);
```
Maybe the naming could be clearer for this? (Present in other places as well if you choose to change it)

blaukc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Jojo :)");
+        System.out.println("What can I do for you?");
+        System.out.println("-----------------------------------------------------------------------");
+        System.out.println("-----------------------------------------------------------------------");
```
Maybe these few lines could be neater if you put all the strings into one `println` call? Like:
```
System.out.println("Hello...\n"
        + "What... \n"
        +...
);
````

blaukc reviewed (commented):
Code generally adheres to good code quality guidelines. Good job!

jocelyn-soh committed:
Add Save.java

jocelyn-soh committed:
Duke.txt: remove unnecessary test cases

jocelyn-soh committed:
Merge branch 'branch-level-7'

jocelyn-soh committed:
Change Deadline.java to LocalDateTime instead of String

jocelyn-soh committed:
Clear duke.txt

jocelyn-soh committed:
Merge branch 'branch-Level-8'

jocelyn-soh committed:
Add TaskList.java, Ui.java, Parser.java and Storage.java

jocelyn-soh committed:
Merge branch 'branch-A-MoreOOP'

jocelyn-soh committed:
Duke.java: Delete unncessary comments

jocelyn-soh committed:
Merge branch 'branch-A-MoreOOP'

jocelyn-soh committed:
Add duke package

jocelyn-soh committed:
Merge branch 'add-gradle-support'

jocelyn-soh committed:
Add JUnit tests

jocelyn-soh committed:
Update test cases

jocelyn-soh committed:
Package app as jar file

jocelyn-soh committed:
Add JavaDocs

jocelyn-soh committed:
Update JavaDoc

jocelyn-soh committed:
Add find method

jocelyn-soh committed:
Fix bugs

jocelyn-soh committed:
Add JavaDoc

jocelyn-soh committed:
Remove wildcard import in Storage.java

jocelyn-soh committed:
Merge branch 'branch-A-Coding-Standard'

jocelyn-soh committed:
Add printFind method

jocelyn-soh committed:
Merge branch 'branch-A-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Ui.java

appleraincoat commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+
+    public Deadline(String description, LocalDateTime by) {
+        super(description);
+        this.by = by;
```
Perhaps you can consider changing variable name "by" to "byWhen"?

appleraincoat commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+            Scanner sc = new Scanner(System.in);
+            Parser.parse(sc, this.ui, this.tasks, this.storage);
+            this.ui.showExitMessage();
+            this.ui.breakLines();
```
"this" is usually only used in constructors, I believe there is no need to use it here 😅

appleraincoat commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,146 @@
+            } else if (cmd.startsWith("mark")) {
+                ui.printMark(tl, cmd);
+            } else if (cmd.startsWith("unmark")) {
+               ui.printUnmark(tl, cmd);
```
Oops, seems like you need 1 more space here!

appleraincoat commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,149 @@
+                        LocalDateTime by = LocalDateTime.parse(splitStr[3].strip(), DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm"));
+                        task = new Deadline(splitStr[2].strip(), by);
+                        break;
+                    case "E":
```
Perhaps you can consider making the case statements on the same indentation level as switch?

appleraincoat commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,53 @@
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
```
I like how you gave the boolean variables names that sound boolean 😄

appleraincoat commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,190 @@
+    /**
+     * Prints an exit message.
+     */
+    public void showExitMessage() {
```
I like your choice of method names throughout the project so far, nice job!

appleraincoat reviewed (commented):
Looks good to me! Besides some small issues, you have adhered to the bulk of the coding style (proper indentation, proper method and variable names) well. Great job!

jocelyn-soh committed:
Add GUI

jocelyn-soh committed:
Revert "Add GUI"

This reverts commit dd73d82c19891c4da6a354309d4a5ace40f104a5.

jocelyn-soh committed:
Revert "Revert "Add GUI""

This reverts commit c8e6fa438da32214fdc29d42b9682022d84131fa.

jocelyn-soh committed:
Fix bugs

jocelyn-soh committed:
Update GUI

jocelyn-soh committed:
Add bye message in GUI

jocelyn-soh committed:
Fix bugs for GUI

jocelyn-soh committed:
Update JavaDoc

jocelyn-soh committed:
Merge branch 'branch-Level-10'

jocelyn-soh committed:
Delete duke.jar

jocelyn-soh committed:
Make bug fixes to the CLI exception handling

jocelyn-soh committed:
Add assertions to methods in Jojo.java,
Parser.java and Ui.java

There are no assertions used in the code.

There is no assumption check at any point in the
code.

Let's add assertions to define assumptions about
the program states.

Adding assertions can help to ensure that the
program follows its expected behaviour.

jocelyn-soh committed:
Replace hard-coded line separator value with
lineSeparator()

Program hard codes the line separator value.

Magic values should be avoided for good code
quality.

Let's replace the hard-coded line separator value
with the in-built lineSeparator() method in System
object.

jocelyn-soh committed:
Simplify nested loops and long methods in code

There are several methods with deep nesting and
long methods.

Deeply nested code makes it hard for the reader
to keep track of the logic while long methods
provide too much information for the reader to
process at a time.

Let's,
* simplify nested loops by breaking parts of the
code up into helper functions
* shorten long methods by writing code more
concisely

Minimally nested loops and short methods
increase readability, which improves code quality.

jocelyn-soh committed:
Merge pull request #2 from jocelyn-soh/branch-A-Assertions

Add assertions to methods in code

jocelyn-soh committed:
Merge branch 'branch-A-Assertions'

jocelyn-soh committed:
Merge branch 'master' into A-CodeQuality

jocelyn-soh committed:
Merge pull request #3 from jocelyn-soh/A-CodeQuality

Improve code quality

jocelyn-soh committed:
Merge branch 'master' of https://github.com/jocelyn-soh/ip

jocelyn-soh committed:
Improve find method to allow for partial and case-insensitive search

jocelyn-soh committed:
Merge branch 'branch-BCD-Extension'

jocelyn-soh committed:
Update error message in JojoUnknownTaskException

jocelyn-soh committed:
Revert "Update error message in JojoUnknownTaskException"

This reverts commit 20b6d0fbd84a7a9224f7a52e825bf7943e12464c.

jocelyn-soh committed:
Merge branch 'branch-A-Assertions'

jocelyn-soh committed:
Update README.md

jocelyn-soh committed:
Rename unmarkTest method in ToDoTest.java

jocelyn-soh committed:
Update Storage.java

jocelyn-soh committed:
Update Storage.java

jocelyn-soh committed:
Merge branch 'branch-A-CodeQuality'

jocelyn-soh committed:
Update GUI

jocelyn-soh committed:
Update ParserTest.java

jocelyn-soh committed:
Update Parser.java

jocelyn-soh committed:
Merge branch 'branch-A-CodeQuality'

jocelyn-soh committed:
Add Ui.png

jocelyn-soh committed:
Update README.md

jocelyn-soh committed:
Update README.md

jocelyn-soh committed:
Update README.md

jocelyn-soh committed:
Update README.md

jocelyn-soh committed:
Update build.gradle

jocelyn-soh committed:
Update Storage.java to create new txt file if it does not exist
